src/analysis/calculus/tangent_cone.lean

```diff
@@ -340,13 +343,36 @@ begin
   exact mem_tangent_cone_of_segment_subset (conv.segment_subset xs zs)
 end
 
+lemma unique_diff_on_Ici (a : ℝ) : unique_diff_on ℝ (Ici a) :=
+unique_diff_on_convex (convex_Ici a) $ by simp only [interior_Ici, nonempty_Ioi]
+
+lemma unique_diff_on_Iic (a : ℝ) : unique_diff_on ℝ (Iic a) :=
+unique_diff_on_convex (convex_Iic a) $ by simp only [interior_Iic, nonempty_Iio]
+
+lemma unique_diff_on_Ioi (a : ℝ) : unique_diff_on ℝ (Ioi a) :=
+is_open_Ioi.unique_diff_on
+
+lemma unique_diff_on_Iio (a : ℝ) : unique_diff_on ℝ (Iio a) :=
+is_open_Iio.unique_diff_on
+
+lemma unique_diff_on_Icc {a b : ℝ} (hab : a < b) : unique_diff_on ℝ (Icc a b) :=
```

Comment:
Do we need `hab`? Or can we do cases on `a < b` and proof the degenerate case using `unique_diff_on_singleton` and `unique_diff_on_empty`?

Reply:
singletons don't have unique derivatives...

Reply:
Aha, I guess that actually makes sense [silly me].